src/bm_sim/P4Objects.cpp

```diff
@@ -2211,9 +2212,20 @@ P4Objects::check_required_fields(
 
 std::unique_ptr<CalculationsMap::MyC>
 P4Objects::check_hash(const std::string &name) const {
-  auto h = CalculationsMap::get_instance()->get_copy(name);
-  if (!h) throw json_exception(EFormat() << "Unknown hash algorithm: " << name);
-  return h;
+  {
+    auto h = CalculationsMap::get_instance()->get_copy(name);
+    if (h) return h;
+  }
+  {
+    auto h = CrcMap::get_instance()->get_copy(name);
+    if (h) return h;
+  }
+  {
+    auto h = CrcMap::get_instance()->get_copy_from_custom_str(name);
```

Comment:
No particular reason, but I don't see a compelling reason to have a single public method as well, especially given that there is a single call site.